utils.go

```diff
@@ -407,6 +407,39 @@ func ReadStringWithMax(r io.Reader, maxLength uint64) (string, error) {
 	return string(buf), nil
 }
 
+// ReadFullStringIntoBuf will read a string off the given stream, consuming the
+// entire cbor item if the string on the stream is longer than the buffer,
+// the string is discarded and 'false' is returned
+func ReadFullStringIntoBuf(cr *CborReader, buf []byte, maxLength uint64) (int, bool, error) {
+	maj, l, err := cr.ReadHeader()
+	if err != nil {
+		return 0, false, err
+	}
+
+	if maj != MajTextString {
+		return 0, false, fmt.Errorf("got tag %d while reading string value (l = %d)", maj, l)
+	}
+
+	if l > maxLength {
+		return 0, false, fmt.Errorf("string in input was too long")
+	}
+
+	if l > uint64(len(buf)) {
+		_, err := io.CopyN(io.Discard, cr, int64(l))
+		if err != nil {
+			return 0, false, err
+		}
+		return 0, false, nil
+	}
+
+	_, err = io.ReadAtLeast(cr, buf[:l], int(l))
```

Comment:
`ReadFull`?

Reply:
hah, yeah, thats ever so slightly cleaner
```
func ReadFull(r Reader, buf []byte) (n int, err error) {
	return ReadAtLeast(r, buf, len(buf))
}
```

Reply:
Ah, I didn't realize that that was the implementation...